warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java

```diff
@@ -444,12 +446,33 @@ public void initialize(ShardQueryConfiguration config, AccumuloClient client, Qu
                 config.setProjectFields(getCardinalityConfiguration().getRevisedProjectFields(queryModel, originalProjectFields));
             }
 
-            this.queries = getQueryPlanner().process(config, jexlQueryString, settings, this.getScannerFactory());
-
+            // If the planner is a DefaultQueryPlanner, delegate the execution to a FederatedQueryPlanner.
+            if (getQueryPlanner() instanceof DefaultQueryPlanner) {
```

Comment:
Why not have the "getQueryPlanner()" simply return a FederatedQueryPlanner ?  This way we can do some A/B testing to ensure we are doing the right thing.  Currently we set the query planner to a configured DefaultQueryPlanner via the QueryLogicFactory.xml.  Perhaps we can set the delegated query planner via a setQueryPlanner on the FederatedQueryPlanner itself.

Reply:
I took this approach initially because I wasn't sure what other logic in ShardQueryLogic would be dependent on having access directly to the original DefaultQueryPlanner instance. I'll switch to your suggested approach and see if I run into any issues.